presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -915,15 +916,23 @@ public void setHiveMetastorePrestoKeytab(String hiveMetastorePrestoKeytab)
         this.hiveMetastorePrestoKeytab = hiveMetastorePrestoKeytab;
     }
 
-    public boolean isHdfsSaslEnabled()
+    enum AuthenticationType
     {
-        return hdfsSaslEnabled;
+        SIMPLE,
+        SIMPLE_IMPERSONIFICATION,
+        KERBEROS,
+        KERBEROS_IMPERSONIFICATION
     }
 
-    @Config("hive.hdfs.sasl.enabled")
-    public void setHdfsSaslEnabled(boolean hdfsSaslEnabled)
+    public AuthenticationType getHdfsAuthenticationType()
     {
-        this.hdfsSaslEnabled = hdfsSaslEnabled;
+        return hdfsAuthenticationType;
+    }
+
+    @Config("hive.hdfs.authentication.type")
+    public void setHdfsAuthenticationType(AuthenticationType hdfsAuthenticationType)
```

Comment:
please rebase on the newer version and update `TestHiveClientConfig` according to your latest changes